contracts/auction/LANDAuction.sol

```diff
@@ -18,31 +18,34 @@ contract LANDAuction is Ownable, LANDAuctionStorage {
     * @param _duration - uint256 duration of the auction in seconds
     * @param _manaToken - address of the MANA token
     * @param _landRegistry - address of the LANDRegistry
+    * @param _dex - address of the Dex to convert ERC20 tokens allowed to MANA
     */
     constructor(
         uint256 _initialPrice, 
         uint256 _endPrice, 
         uint256 _duration, 
-        address _manaToken, 
-        address _landRegistry
-    ) public 
-    {
-        require(
-            _manaToken.isContract(),
-            "The MANA token address must be a deployed contract"
-        );
-        manaToken = MANAToken(_manaToken);
+        ERC20 _manaToken, 
+        LANDRegistry _landRegistry,
+        address _dex
+    ) public {
+        Ownable.initialize(msg.sender);
 
         require(
-            _landRegistry.isContract(),
+            address(_landRegistry).isContract(),
             "The LANDRegistry token address must be a deployed contract"
         );
-        landRegistry = LANDRegistry(_landRegistry);
+        landRegistry = _landRegistry;
+
+        if (_dex != address(0)) {
+            setDex(_dex);
+        }
+
+        allowToken(address(_manaToken), 18, true);
+        manaToken = _manaToken;
 
         require(_initialPrice > _endPrice, "The start price should be greater than end price");
```

Comment:
Maybe we could move this requires to the start of the function, under the first require()

Reply:
I removed the `_initialPrice` and `_endPrice` in #16